dev_guide/builds/build_inputs.adoc

```diff
@@ -0,0 +1,911 @@
+[[dev-guide-build-inputs]]
+= Build Inputs
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[how-build-inputs-work]]
+== How Build Inputs Work
```

Comment:
Re-titled "Providing Source for a Build" -> "How Build Inputs Work", since "Providing" made it seem like there was actually a task in here, which there is not.